Automattic/jetpack
# Make "My Jetpack" the default page in Jetpack wp-admin (2nd take).

Reverts Automattic/jetpack#32384

This diff makes **My Jetpack** the default page that shows when the top-level "**Jetpack**" menu item is clicked in the left sidebar in WP Admin. Also make "My Jetpack" the first sub-menu nav item.

It was agreed to split this change into two pieces:
1. First we just make My Jetpack the default with no other changes. (This PR).
2. Then, we will bring more dynamic data into My Jetpack, and over time at some point we get rid of At a Glance.

We're hoping to get this in before the _Aug 16th code freeze_. Please see: p1690031014020359/1690029940.688399-slack-CDLH4C1UZ

Per _James Grierson_ in: **Unleashing Our Potential: A Roadmap to Flourish in the Upcoming Months**: p1HpG7-mMC-p2

> **My Jetpack as the default in August** –  My Jetpack is the page that allows users to see and explore our other products. Now that we have unified the experience of our plugins, we need to make My Jetpack the top level starting point when someone clicks on Jetpack in the WP-Admin. Doing so will increase exploration and the flow of new users to our individual plugins.






Project Thread: pbNhbs-7IV-p2

## Proposed changes:

- Makes the "Jetpack" menu item link to `?page=my-jetpack` instead of `?page=jetpack#/dashboard`
- Moves the "My Jetpack" sub-menu item to first position, followed by each activated standalone plugin (if disconnected), or followed by "Dashboard", "Settings", and each activated plugin (if connected). 
- Updates the jQuery that controls which sub-menu item is currently active/highlighted to account for the re-positioned "My Jetpack" sub-menu link.
- Updates connection e2e tests to account for the top-level "Jetpack" menu item now pointing to the My Jetpack page.

### Other information:


## Jetpack product discussion



p1HpG7-mMC-p2#comment-63646

p1690029940688399-slack-CDLH4C1UZ


## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Do one of these:
    - Checkout this PR and spin up a local development build of Jetpack, OR
    - Spin up a Jurassic.ninja site with the Jetpack Beta plugin running this PR.
- Jetpack plugin activated, Not connected:
    - Click the top-level "Jetpack" menu item and verify it takes you to the "My Jetpack" page.
    - Verify "My Jetpack" is the first sub-item in the Jetpack menu, with any activated plugins following suit.
    - Verify the currently active menu-item is highlighted correctly. ie.- When on My Jetpack page, the My Jetpack menu item is highlighted.

Jetpack plugin activated, **not connected**:
**BEFORE PR** | **AFTER PR**
--- | ---
![Markup 2023-08-02 at 14 18 50](https://github.com/Automattic/jetpack/assets/11078128/9461b192-d47e-430f-8a2c-e0cb7d9f9ebc) | ![Markup 2023-08-02 at 14 22 28](https://github.com/Automattic/jetpack/assets/11078128/6d9f6e15-04b3-410d-9a17-9476fc4c6bba)

- Connect Jetpack.
    - Click the top-level "Jetpack" menu item and verify it takes you to the "My Jetpack" page.
    - View the Jetpack side menu. Verify "My Jetpack" is the first menu item, with "Dashboard" below it, "Settings" below that, and with any activated plugins following suit.
    - Verify the "My Jetpack" sub-menu item takes you to the "My Jetpack" page, and the "Dashboard" sub-menu item takes you to the Jetpack "Dashboard" page.
    - Verify the currently active menu-item is highlighted correctly. ie.- When on the "My Jetpack" page, the "My Jetpack" sub-menu item is active/highlighted, and when on the "Dashboard" page, the "Dashboard" sub-menu item is highlighted.

Jetpack **connected**:
**BEFORE PR** | **AFTER PR**
--- | ---
![Markup 2023-08-02 at 14 25 57 (1)](https://github.com/Automattic/jetpack/assets/11078128/0381a89d-5e2b-49e2-b86b-d5aecc4bce51) | ![Markup 2023-08-02 at 14 28 11](https://github.com/Automattic/jetpack/assets/11078128/49711ed8-596e-4954-bdad-093de5ccae96)

- Non-admin user:
    - Create a user with role Contributor and log into site with that user.
    - Visit wp-admin
    - Confirm the Jetpack -> My Jetpack menu item is shown.
    - Confirm the My Jetpack page loads without error. (See: https://github.com/Automattic/jetpack/pull/32420)
